woocommerce/wc-admin
# Add optimistic task completion and cache invalidation

Fixes #7371 

The original issue involved watching other data stores and invalidating cache, however, I think foregoing this for now is the best option because:
1. We don't have any cases where a task can be completed outside the task list without a page refresh.  We probably will in the future, but there is no testable case yet.
2. We don't have a great way to watch tasks.  I realized the proposed solution in #7371 would only work if the task list is mounted and this is not ideal.

This PR instead updates task optimistically to show as completed and verifies this by invalidating cache of the tasks REST API.

### Detailed test instructions:

1. Open your network tab.
2. Walk through each task, completing it.
3. Note the task is immediately marked as complete and navigates back to the task list (except for the marketing task)
4. Check that after task completion the `tasks` endpoint is called again.
5. Optionally, adjust the `is_complete` logic in a task on the server (e.g., `src/Features/OnboardingTasks/Tasks/Appearance.php` ) to `false` and complete it client-side to note that it is reverted to incomplete after optimistically marking as complete.
